crates/storage/src/lazy/vec.rs

```diff
@@ -0,0 +1,677 @@
+    /// * `Some(Err(_))` if the value existed but its length exceeds the static buffer
+    ///   size.
+    /// * `None` if there was no value at `index`.
+    pub fn try_get(&self, index: u32) -> Option<ink_env::Result<V>> {
+        self.elements.try_get(index)
+    }
+
+    /// Set the `value` at given `index`.
+    ///
+    /// # Panics
+    ///
+    /// * If the index is out of bounds.
+    /// * If decoding the element exceeds the static buffer size.
+    pub fn set<T>(&mut self, index: u32, value: &T) -> Option<u32>
+    where
+        T: Storable + EncodeLike<V>,
+    {
+        assert!(index < self.len());
+
+        self.elements.insert(index, value)
+    }
+
+    /// Try to set the `value` at given `index`.
+    ///
+    /// Returns:
+    ///
+    /// * `Ok(Some(_))` if the value was inserted successfully, containing the size in
+    ///   bytes of the pre-existing value at the specified key if any.
+    /// * `Ok(None)` if the insert was successful but there was no pre-existing value.
+    /// * Err([ink_env::Error::BufferTooSmall]) if the encoded value exceeds the static
+    ///   buffer size
+    /// * Err([ink_env::Error::KeyNotFound]) if the `index` is out of bounds.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `index` exceeds the length of the vector.
+    pub fn try_set<T>(
+        &mut self,
+        index: u32,
+        value: &T,
+    ) -> Result<Option<u32>, ink_env::Error>
+    where
+        T: Storable + EncodeLike<V>,
+    {
+        if index >= self.len() {
+            return Err(ink_env::Error::KeyNotFound);
+        }
+
+        self.elements.try_insert(index, value)
+    }
+
+    /// Delete all elements from storage.
+    ///
+    /// # Warning
+    ///
+    /// This iterates through all elements in the vector; complexity is O(n).
+    /// It might not be possible to clear large vectors within a single block!
+    pub fn clear(&mut self) {
+        for i in 0..self.len() {
+            self.elements.remove(i);
+        }
+        self.set_len(0);
+    }
+
+    /// Remove the element at `index`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `index` exceeds the length of the vector.
+    pub fn clear_at(&mut self, index: u32) {
+        assert!(index < self.len());
+
+        self.elements.remove(index);
+    }
+}
+
+impl<'a, V, KeyType> From<&'a [V]> for StorageVec<V, KeyType>
+where
+    V: Packed + EncodeLike<V>,
+    KeyType: StorageKey,
+{
+    fn from(value: &'a [V]) -> Self {
+        let mut result = StorageVec::<V, KeyType>::new();
+
+        for element in value {
+            result.push(element);
+        }
+
+        result
+    }
+}
+
+impl<V, KeyType> ::core::fmt::Debug for StorageVec<V, KeyType>
+where
+    V: Packed,
+    KeyType: StorageKey,
+{
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        f.debug_struct("StorageVec")
+            .field("key", &KeyType::KEY)
+            .field("len", &self.len)
```

Comment:
For debugging we may want the `len_cached` too

Reply:
Added it